tests/scripts/pkgconfig.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+#
+# Purpose
+#
+# Test pkgconfig files.
+#
+# For each of the build pkg-config files, .pc files, check that
+# they validate and do some basic sanity testing on the output,
+# ie. that the strings are non-empty.
+#
+# NOTE: This requires the built pc files to be on the pkg-config
+# search path, this can be controlled with env variable
+# PKG_CONFIG_PATH. See man(1) pkg-config for details.
+#
+
+set -e -u
+
+all_pcs="mbedtls mbedx509 mbedcrypto"
+
+for pc in $all_pcs; do
+    echo "testing package config file: ${pc} ... " | tr -d '\n'
```

Comment:
I'd normally use `printf %s "whatever"` for maximum shell compatibility without needing extra programs, or `echo -n` or `echo "xxx\\c"` if I know the shell. Not sure how others feel about this, so that is just a comment